src/cares_wrap.cc

```diff
@@ -1096,78 +1102,71 @@ int ParseSoaReply(Environment* env,
     const int rr_len = cares_get_16bit(ptr + 8);
     ptr += NS_RRFIXEDSZ;
 
-    /* only need SOA */
+    // only need SOA
     if (rr_type == ns_t_soa) {
-      ares_soa_reply soa;
+      char* nsname_temp;
+      char* hostmaster_temp;
 
-      status = ares_expand_name(ptr, buf, len, &soa.nsname, &temp_len);
+      status = ares_expand_name(ptr, buf, len, &nsname_temp, &temp_len);
```

Comment:
Same here

Reply:
@refack Currently the status to return is captured in one variable 'status'. In a for loop, all the 'break' statements breaks out of loop and return some status. Having multiple of these small scoped variables breaking the flow by having multiple return statements instead of break in a loop.  I've not seen any explicit reference, but is this acceptable coding standard?

Reply:
IMHO what you did is great, that's exactly one of the reasons to avoid recycling.
Now you stop the function on first error and propagate it, and not continue to process in an error state.
Double win 🥇

Reply:
BTW: it's was simpler to do this now because of the RAII semantics of ares_unique_ptr.
Triple win 🏆